### PR TITLE
Implementing / Refactoring Events

### PR DESCRIPTION
**Pull request by dredhorse**
Signed-off-by: Don Redhorse dredhorse@breiden.net

Before pulling please discuss: http://forums.spout.org/threads/cause-and-event-system.6129

### CODE REVIEW

File: src/main/java/org/spout/vanilla/event/block/FurnaceSmeltEvent.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	/**
+	 * Sets the result ItemStack
+	 * @param newResult the result ItemStack
+	 */
+	public void setResult(ItemStack newResult) {
+		result = newResult;
+	}
+
+	/**
+	 * Returns the {@link org.spout.vanilla.event.cause.misc.FurnaceSmeltCause} which caused the FurnaceSmeltEvent
```
Comment (kitskub) [inline src/main/java/org/spout/vanilla/event/block/FurnaceSmeltEvent.java]: Nah-ah! Look at the construtor. Also look t the javdocs for the class, same as FurnaceBurnEvent

Comment (dredhorse) [inline src/main/java/org/spout/vanilla/event/block/FurnaceSmeltEvent.java]: yeah  copy paste error... will fix that if this is the way we go further, thanks for catching it...

**Comment by dredhorse** [comment]
I just found out that my furnace causes suck... because a furnace doesn't has any reference to the block he is, so I will change that in a sec too.

**Comment by kitskub** [comment]
@dredhorse : getOwner()?

**Comment by dredhorse** [comment]
sigh.. you are right.. thanks

**Comment by windy1** [comment]
It doesn't have a reference to the block, just the position, and that is intentional.

**Pull request merged**